pandas/tests/internals/test_internals.py

```diff
@@ -1238,7 +1238,6 @@ class TestCanHoldElement(object):
         (2**63, 'complex128'),
         (True, 'bool'),
         (np.timedelta64(20, 'ns'), '<m8[ns]'),
-        (np.datetime64(20, 'ns'), '<M8[ns]'),
```

Comment:
Why removed?  (Besides no is exempt sub working)?

Reply:
This PR lead to `datetime64 OP datetime64 = timedelta64`, regardless of `OP`. Although meaningless, it will result `datetime64 + datetime64 = timedelta64`. 

The assumption `datetime64 OP datetime64 = datetime64` is meaningless for all `OP`. If allow this line, it will require extra `OP` check in `DatetimeBlock.eval`, which I don't think worth.